src/app/modules/main/wallet_section/activity/controller.nim

```diff
@@ -122,25 +126,38 @@ QtObject:
             error "failed to find pending transaction with identity: ", identity
           ptIndex += 1
 
-  proc refreshData(self: Controller) =
+  proc loadingDataChanged*(self: Controller) {.signal.}
+
+  proc setLoadingData(self: Controller, loadingData: bool) =
+    self.loadingData = loadingData
+    self.loadingDataChanged()
 
-    # result type is RpcResponse
-    let response = backend_activity.getActivityEntries(self.addresses, self.chainIds, self.currentActivityFilter, 0, 10)
-    # RPC returns null for result in case of empty array
-    if response.error != nil or (response.result.kind != JArray and response.result.kind != JNull):
+  proc errorCodeChanged*(self: Controller) {.signal.}
+
+  proc setErrorCode(self: Controller, errorCode: int) =
+    self.errorCode = backend_activity.ErrorCode(errorCode)
+    self.errorCodeChanged()
+
+  proc refreshData(self: Controller) =
```

Comment:
do we want to avoid triggering a new filter task if we're currently loading? (not sure if it's done elsewhere, but we might want to add a protection here)
If we don't mind running several filter tasks at the same time, then we should wait for the last one to finish before setting `setLoadingData` to false

Reply:
This is part of the presentation layer; therefore, QML will handle this responsibility based on the UX state. You can have a look at the workflow I demo in the debugging `ActivityView` from the last PR: https://github.com/status-im/status-desktop/blob/b1089921faade36ecb9e6f3f61dca5f579e623b9/ui/imports/shared/views/ActivityView.qml

Main workflows:
- The user lands on the activity history the QML call `updateFilter`
- Any other parallel call is avoided with `loadingData` property time in which the loading animation runs
- After data comes in, no more loadingData is needed `loadMoreItems` is called which will get the controller and QML in the previous state

@Cuteivist is looking into the final integration, and he can judge fi this is a good workflow between QML and Controller. Also change it if it doesn't serve user needs. The current implementation works for the provided debugging view.